### ISSUE TITLE
Bug: `allowConstructorFlags` option of `no-invalid-regexp` allows duplicate strings

### ISSUE THREAD

**Issue by Tanujkanti4441** [eslint/eslint#18755]
### Environment

Node version: latest
npm version: latest
Local ESLint version: latest
Global ESLint version: no
Operating System: windows

### What parser are you using?

Default (Espree)

### What did you do?

</details>

```js
/*eslint no-invalid-regexp: ["error", { "allowConstructorFlags": ["a", "a"] }]*/

new RegExp('.', 'a');
```

### What did you expect to happen?

eslint will report duplicate flag passed to `allowConstructorFlags`.

### What actually happened?

eslint doesn't report it

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8taW52YWxpZC1yZWdleHA6IFtcImVycm9yXCIsIHsgXCJhbGxvd0NvbnN0cnVjdG9yRmxhZ3NcIjogW1wiYVwiLCBcImFcIl0gfV0qL1xuXG5uZXcgUmVnRXhwKCcuJywgJ2EnKTsiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
per the versioning-policy, the fix could be a breaking change  (rule schema changes): https://github.com/eslint/eslint?tab=readme-ov-file#semantic-versioning-policy

We can table it for v10. thoughts? @eslint/eslint-team

**Comment by JoshuaKGoldberg** [comment]
Agreed all around: both that this is would be a good change, and that it's technically a breaking change. There's no reason why anybody would want duplicate strings in the rule options.

**Comment by mdjermanovic** [comment]
> per the versioning-policy, the fix could be a breaking change (rule schema changes): https://github.com/eslint/eslint?tab=readme-ov-file#semantic-versioning-policy
> 
> We can table it for v10. thoughts?

Makes sense to me.

**Comment by sam3k** [comment]
TSC group has confirmed this for `v10`

**Comment by mdjermanovic** [comment]
@Tanujkanti4441 We have started preparing PRs for ESLint v10. Would you like to submit a PR for this change?

**Comment by Tanujkanti4441** [comment]
> @Tanujkanti4441 We have started preparing PRs for ESLint v10. Would you like to submit a PR for this change?

Yes, i will send a PR for this.
